tests/unit/lib/sftp-tests.js

```diff
@@ -1,10 +1,154 @@
 const EventEmitter = require("events");
 const {expect} = require("chai");
 const {Future, fork} = require("fluture");
+const {getFileViaSftp, sendFileViaSftp} = require("../../../lib/sftp.js");
 const {Readable, PassThrough} = require("stream");
-const {sendFileViaSftp} = require("../../../lib/sftp.js");
 
 describe("Unit Tests - SFTP", function() {
+  describe("getFileViaSftp", function() {
+    it("should return a future", function() {
+      const fakeConnectionConfig = {
+        "host": "",
+        "port": 1,
+        "remoteFilePath": "",
+        "user": "",
+        "password": "",
+      };
+
+      // this is required for sanctuary type checking
+      const mockSftpClient = {
+        "connect": () => {},
+        "end": () => {},
+        "on": () => {},
+        "sftp": () => {},
+      };
+
+      expect(getFileViaSftp(mockSftpClient)(fakeConnectionConfig)).to.be.instanceOf(Future);
+    });
+
+    it("should resolve with a readstream if get succeeds", function(done) {
+      const fakeConnectionConfig = {
+        "host": "",
+        "port": 1,
+        "remoteFilePath": "some_file.txt",
+        "user": "",
+        "password": "",
+      };
+
+      const readable = new Readable();
+      const mockSftpClient = new EventEmitter();
+
+      const sftp = {
+        "createReadStream": (remoteFilePath) => {
+          return readable;
+        },
+      };
+
+      mockSftpClient.sftp = (cb) => {
+        cb(null, sftp);
+      };
+      mockSftpClient.connect = () => {
+        mockSftpClient.emit("ready");
+      };
+      mockSftpClient.on("ready", () => {});
+      mockSftpClient.end = () => {};
+
+      fork
+      (done)
+      (data => {
+        let result = "";
+
+        data.on("data", function(d) {
+          result += d.toString();
+        });
+
+        data.on("end", function() {
+          expect(result).to.deep.equal("hello world");
+          done();
+        });
+      })
+      (getFileViaSftp(mockSftpClient)(fakeConnectionConfig));
+
+      readable.push("hello world");
+      readable.push(null);
+    });
+
+    it("should reject if the get fails", function(done) {
```

Comment:
```suggestion
    it("should reject if the 'get' method fails", function(done) {
```
Or something like this to make it more clear.